spec/services/bulk_upload/sales/year2023/row_parser_spec.rb

```diff
@@ -651,6 +651,26 @@
         end
       end
     end
+
+    describe "soft validations" do
+      context "when soft validation is triggered" do
+        let(:attributes) { valid_attributes.merge({ field_30: 22, field_35: 5 }) }
+
+        it "adds an error to the relevant fields" do
+          soft_validation_errors = parser.errors.select { |e| e.options[:category] == :soft_validation }
+
+          expect(soft_validation_errors.find { |e| e.attribute == :field_30 }).to be_present
```

Comment:
i think i found a slightly nicer way to write this which I've started using
```suggestion
          expect(parser.errors.where(:field_30, category: :soft_validation)).to be_present
```

Reply:
Very nice, thanks!